src/context/subscription.tsx

```diff
@@ -25,19 +26,27 @@ export const SubscriptionProvider: React.FC<{ children: ReactNode }> = ({ childr
 
   const fetchSubscription = useCallback(async () => {
     if (token && profile) {
-      const subscription = await getSubscriptionById(token, profile.id);
-
-      setSubscription({
-        active: subscription.active,
-        invoice_pdf: subscription.invoice_pdf,
-        current_period_end: subscription.current_period_end,
-        current_period_start: subscription.current_period_start,
-        product_name: subscription.product_name,
-        cancel_at_period_end: subscription.cancel_at_period_end
-      });
+      try {
+        // Attempt to fetch the subscription, handling the case where no subscription exists
+        // This could mean the user hasn't purchased a subscription or the subscription was cancelled
+        const subscription = await getSubscriptionById(token, profile.id);
+        setSubscription({
+          active: subscription.active,
+          invoice_pdf: subscription.invoice_pdf,
+          current_period_end: subscription.current_period_end,
+          current_period_start: subscription.current_period_start,
+          product_name: subscription.product_name,
+          cancel_at_period_end: subscription.cancel_at_period_end
+        });
+      } catch (error) {
+        if (axios.isAxiosError(error) && error.response?.status === 404) {
+          setSubscription(defaultSubscriptionData);
+          return;
```

Comment:
We may not even need to do the set since it's already the default subscription when we initialize state, but it doesn't harm to have it either.
Also we could take the return out.
Feel free to address or not I don't mind both version are fine, one is shorter, the other is more explicit